AppCenter/AppCenter/Internals/Util/MSUtility+StringFormatting.m

```diff
@@ -40,10 +52,38 @@ + (NSString *)appSecretFrom:(NSString *)string {
     return nil;
   } else {
     for (NSString *component in components) {
-
+      BOOL transmissionTokenIsNotPresent = [component rangeOfString:kMSTransmissionTargetKey].location == NSNotFound
+      && [component rangeOfString:kMSTransmissionIosTargetKey].location == NSNotFound
+      && [component rangeOfString:kMSTransmissionMacosTargetKey].location == NSNotFound;
+      
       // Component is app secret, return the component. Check for length > 0 as "foo;" will be parsed as 2 components.
-      if (([component rangeOfString:kMSTransmissionTargetKey].location == NSNotFound) && (component.length > 0)) {
-        NSString *secretString = [component stringByReplacingOccurrencesOfString:kMSAppSecretKey withString:@""];
+      if (transmissionTokenIsNotPresent && (component.length > 0)) {
+        NSString *secretString = @"";
+        
+        if ([string rangeOfString:kMSAppSecretKey].location == NSNotFound) {
```

Comment:
One suggestion is we can start with the most specific key so `ios` or `macos` then `appsecret` to be slightly more robust to some error case for instance `ios=<..>;macos=<..>;appsecret=<..>;` or `ios=<..>;macos=<..>;<..>;` will set the right appsecret for iOS and macOS.